### ISSUE TITLE
Android 15 (API 35) edge-to-edge causes breakage

### ISSUE THREAD

**Issue by yawnoc** [stroke-input/stroke-input-android#17]
665328a2645a444b4b47c35372423f26d962bbf5 causes breakage pertaining to edge-to-edge enforcement.

- https://developer.android.com/about/versions/15/behavior-changes-15#edge-to-edge
- https://old.reddit.com/r/androiddev/comments/1c1rs1t/apps_targeting_android_15_will_be_forced_into/

| Use case | Android 14 screenshot | Android 15 screenshot | Android 15 breakage |
| - | - | - | - |
| (Portrait) Main Activity: load keyboard by focusing input box | ![Screenshot_20240721_162525](https://github.com/user-attachments/assets/2f6ef45a-01ab-4835-aa89-a81528baa954) | ![Screenshot_20240721_162555](https://github.com/user-attachments/assets/9087e248-985c-46bb-a63a-716d7c6cde10) | Keyboard obscures input box because Main Activity is not scrolled upward. Navigation bar obscures bottom of keyboard. |
| (Portrait) Main Activity: scroll to bottom after focusing input box | ![Screenshot_20240721_162831](https://github.com/user-attachments/assets/12269572-8d92-40a5-8e22-e4ba33415e14) | ![Screenshot_20240721_162853](https://github.com/user-attachments/assets/5f1d1e3c-8874-4130-a047-f3c7a382347d) | Keyboard obscures bottom of Main Activity, which cannot be scrolled upward past the keyboard. |
| (Landscape) Main Activity: load keyboard by focusing input box | ![Screenshot_20240721_163641](https://github.com/user-attachments/assets/7ce9fc67-79e1-444b-8d71-f3c6476da637) | ![Screenshot_20240721_163658](https://github.com/user-attachments/assets/49173850-ebc4-4377-b7c9-aef54498e605) | Navigation bar obscures right edge of keyboard. |
| (Portrait) Main Activity Help | ![Screenshot_20240721_164427](https://github.com/user-attachments/assets/04f3e40c-6398-41ab-9aa5-e9a9929ec52b) | ![Screenshot_20240721_164441](https://github.com/user-attachments/assets/f8952f07-579b-432b-b317-b8cb394bea8a) | Status bar / self-facing camera obscures heading. Navigation bar obscures Return button. |

In summary, cursed be Android 15 edge-to-edge. And cursed be Google.

**Comment by yawnoc** [comment]
So it turns out you can opt out of edge-to-edge:

- Documentation: https://developer.android.com/reference/android/R.attr#windowOptOutEdgeToEdgeEnforcement
- Article: https://www.androidauthority.com/android-15-edge-to-edge-opt-out-3467646/
- Example code: https://medium.com/androiddevelopers/insets-handling-tips-for-android-15s-edge-to-edge-enforcement-872774e8839b (under "Need more time to migrate?")
- Reddit discussion: https://old.reddit.com/r/androiddev/comments/1ga3nkg/android_15_is_supposed_to_force_apps_to_go/

**Comment by yawnoc** [comment]
In summary, 3c84290448ee9cd364527ce2ac621b02e42adee9 does the following:
- Implements keyboard edge-to-edge padding (by programmatically setting the height of a bottom spacer that has been added below the keyboard)
- Opts out of Main Activity edge-to-edge (because cursed be Google)
